PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/KustoQueriesWithTimeIntervalInQueryStringCheck.java

```diff
@@ -0,0 +1,255 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiExpressionList;
+import com.intellij.psi.PsiLiteralExpression;
+import com.intellij.psi.PsiLocalVariable;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiPolyadicExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.PsiVariable;
+import com.intellij.psi.impl.source.tree.java.PsiMethodCallExpressionImpl;
+
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * This class is an inspection tool that checks for Kusto queries with time intervals in the query string.
+ * This approach makes queries less flexible and harder to troubleshoot.
+ * This inspection tool checks for the following anti-patterns:
+ * <ul>
+ * <li> Queries that use the "ago" function with a time interval.</li>
+ * <li> Queries that use the "datetime" function with a specific datetime.</li>
+ * <li> Queries that use the "now" function to get the current timestamp.</li>
+ * <li> Queries that use the "startofday", "startofmonth", or "startofyear" functions.</li>
+ * <li> Queries that use the "between" function with datetime values.</li>
+ * </ul>
+ * <p>
+ * When the anti-patterns are detected as parameters of Azure client method calls, a problem is registered.
+ */
+public class KustoQueriesWithTimeIntervalInQueryStringCheck extends LocalInspectionTool {
+
+    /**
+     * This method builds the visitor for the inspection tool
+     *
+     * @param holder     ProblemsHolder is used to register problems found in the code
+     * @param isOnTheFly boolean to indicate if the inspection is done on the fly -
+     * @return PsiElementVisitor
+     */
+    @NotNull
+    @Override
+    public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new KustoQueriesVisitor(holder, isOnTheFly);
```

Comment:
Don't need to pass `isOnTheFly` as this is anyway ignored. We can just have a single-arg constructor of KustoQueriesVisitor.